sdks/python/apache_beam/io/filebasedsink.py

```diff
@@ -309,11 +341,18 @@ def _template_to_format(shard_name_template):
                        shard_name_template)
     shard_name_format = shard_name_template.replace(
         m.group(0), '%%(shard_num)0%dd' % len(m.group(0)))
-    m = re.search('N+', shard_name_format)
-    if m:
-      shard_name_format = shard_name_format.replace(
-          m.group(0), '%%(num_shards)0%dd' % len(m.group(0)))
-    return shard_name_format
+    return FileBasedSink._template_replace_num_shards(shard_name_format)
+
+  @staticmethod
+  def _template_to_glob_format(shard_name_template):
+    if not shard_name_template:
+      return ''
+    m = re.search('S+', shard_name_template)
```

Comment:
This will restrict us to shard name patterns of the form S+. Can we log a warn and not perform the file deletion instead of failing ?

Reply:
This is not a new requirement: if you look at the code for `_template_to_format` you'll see that it also requires S+ in the pattern. The shard number (S+) is required, while the number of total shards (N+) is optional.